docs: provide general arcus preview (#204)
arcus-azure/arcus: preview/arcus-general.md

# Arcus - Azure development in a breeze.

## What is Arcus?
A set of Codit-developed OSS libraries/tools that simplify building apps running on Microsoft Azure.
It is Codit’s strategy to centralize company-wide best practices and use it across all our customers.

* Built By Experts, For Experts
  Sponsored by Codit! Maintained by a dedicated set of Codit experts that assures quality for our current and future clients.

* Modular Design
  All Arcus components are highly functional and composable. They can be used separately, but they are better together.

* Open-Source & Free to use

>   The Arcus framework is constantly updated by contributions from Codit employees and abroad!

Reviewer: It is true that most (if not all) contributions are made by Codit employees, but I'd say we're open for contributions by anyone.

Author: That's the `...and abroad` part, no? Or would you be more explicit about that?

Reviewer: Yes, I would be more explicit.
Something like:

```suggestion
  The Arcus library is constantly updated by contributions from Codit employees and abroad!  
  As the Arcus library is free to use and open-source so we happily accept your contributions!  Please take a look at our [contribution guide](https://github.com/arcus-azure/.github/blob/master/CONTRIBUTING.md) to get up to speed if you want to contribute.
```
(Are we talking about a framework or a library; imho, Arcus is a library and not a framework).

Author: Librarie(s), 😄 , but yeah, is good for me.

  It is free to use under the MIT license.

> 🚩 Are you doing the same thing at multiple customers? Then we are happy to hear! [Link to arcus GitHub]()

Reviewer: What link should be put here ?

Author: Well, it would be nice if we have a form to link to or something? Whether this form is on GitHub, that would probably be difficult.

Reviewer: And what should that form be about actually ?  I mean, what kind of questions / survey ?
It's not very clear to me what we want to achieve with that form.

Author: We previously had some sort of a questionnaire to ask people about what components they are using, their experience with those components, and/or ideas/suggestions or something. If we get a 'go' from management on the 'customer approval', we can also ask them on which customer projects they use Arcus.

## Why should you use Arcus?

> There exists many reasons to use Arcus in your current and future projects.

Reviewer 2: ```suggestion
There are many reasons to use Arcus in your current and future projects.
```

* Time-saving
  Arcus contains tremendous amounts of boilerplate code that helps you save loads of time during project development.

>   It helps with infrastructure like secure secret management, application-wide telemetry tracking, tedious DevOps operations. and even has fully development-ready project templates to get you started immediately.

Reviewer 2: ```suggestion
  It helps with infrastructure like secure secret management, application-wide telemetry tracking, tedious DevOps operations and even has fully development-ready project templates to get you started immediately.
```

* Centralized

>   The entire Arcus framework is maintained and tested the same way, with the same set of dedicated experts. There exists a single point of truth were everything is maintained and made ready for your benefit.

Reviewer 2: ```suggestion
  The entire Arcus framework is maintained and tested the same way, with the same set of dedicated experts. There exists a single point of truth where everything is maintained and made ready for your benefit.
```

* Best practices

>   Every Arcus component has the company-wide best practices backed right into the code. It is secure by-default, tracks telemetry by itself, automates hard-to-setup DevOps tasks, and many more.

Reviewer 2: maybe mention `industry-wide` instead of `company-wide` best practices. If we use `company-wide` we might fixate it too much on Codit?

Author: https://github.com/arcus-azure/arcus/pull/204/commits/47f67df39c795631893d67f15bd0823ae6dae3f2

> ## How to use Arcus?

Reviewer 2: The list below is more of a `When to use Arcus?` instead of `How to use Arcus?`
Maybe we should also mention this is not a complete list and are examples of usage. For example Arcus.Scripting doesn't necessarily have to be used from a DevOps pipeline 😉

Author: Yes, you're right. Was experimenting with this as I wanted something to get people started after they are convinced by the previous two titles 😅 . I think the 'How to use it' is till relevant but we should maybe change the structure of this. Maybe more focus on the project templates as they are the starting point. But I'm thinking how we will introduce ML and Scripting to this.

And yeah, we should make sure that we mention that this is by all means not 'complete'.

Reviewer 2: I think the current setup is a good one, `What?` to `Why?` to `How?` seems like a good walk-through 👍🏻 
Or maybe use `How and When?`?

The Arcus components are available on NuGet, PowerShell Gallery and Python PyPi.

Are you using Web API technology? Take a look at these components:
* Arcus Web API project template

> * Arcus Azure Functions HTTP trigger

Reviewer 2: ```suggestion
* Arcus Azure Functions HTTP trigger template
```

* Arcus Web API

Are you using messaging technology with Azure? Like Azure Service Bus? Take a look at these components:
* Arcus Azure Service Bus Worker templates
* Arcus Messaging

Are you using repeated background work? Take a look at at these components:
* Arcus Databricks Metrics project template
* Arcus Background Jobs

Are you using Azure resources in your DevOps pipeline? Take a look at these components:
* Arcus Scripting

Are you using ML technology? Take a look at these components:
* Arcus ML

> * Arcus Azure ML

Reviewer: Should we provide links to the github page of every component here ?

Author: Yeah, the documentation page of every comment seems find, I think. Documentation > GitHub.

For more information on all the different Arcus components: [separate page]().

> ## Who's using Arcus?

Reviewer 2: If I remember correctly there was a more extensive list in one of the Powerpoints from some time ago, I can't find it at the moment but it contained a list of customers and an amount of stars for each Arcus component they used.

Author: Ah, yeah, let's add that! 👍

Author: https://github.com/arcus-azure/arcus/pull/204/commits/47f67df39c795631893d67f15bd0823ae6dae3f2

Reviewer 2: Now that I think about it, I'm not sure all these customers have agreed to be listed on the website. 
I did have agreement from Van Gelder to have them listed, but not sure about all the others.

Author: Ah, yeah, maybe the PP we got from Massimo is for internal use only?

Reviewer 2: Yeah If I remember correctly that was the case. Maybe remove the customers you added in https://github.com/arcus-azure/arcus/commit/47f67df39c795631893d67f15bd0823ae6dae3f2 and check with Massimo if these customers can be listed or not?

Author: Send him a message a reverted it https://github.com/arcus-azure/arcus/pull/204/commits/da02b1358afedec24e30fa67ddbbf294b7f79e1f

Arcus is being using in production by these and many more end-users:

* Codit
* Kubernetes event grid bridge
* Promitor

> * VanGelder

Reviewer 2: ```suggestion
* Van Gelder groente & fruit 
```
